nixos/modules/services/databases/cassandra.nix

```diff
@@ -39,15 +42,42 @@ let
         mkdir -p "$out"
 
         echo "$cassandraYaml" > "$out/cassandra.yaml"
-        ln -s "$cassandraEnvPkg" "$out/cassandra-env.sh"
         ln -s "$cassandraLogbackConfig" "$out/logback.xml"
+
+        cp "$cassandraEnvPkg" "$out/cassandra-env.sh"
+
+        # Delete default JMX Port, otherwise we can't set it using env variable
+        sed -i '/JMX_PORT="7199"/d' "$out/cassandra-env.sh"
```

Comment:
If the source line ever changes, this will just fail silently?

Reply:
If they remove or change this line, sed will not remove it and not issue a warning.
`substituteInPlace` would warn, but it doesn't have an option to delete an entire line.

I think it'll be fine if I implement the test for `jmxPort` like you suggested below.